wrappers/hnyecho/echo.go

```diff
@@ -0,0 +1,80 @@
+package hnyecho
+
+import (
+	"sync"
+
+	"github.com/honeycombio/beeline-go/wrappers/common"
+	"github.com/labstack/echo"
+)
+
+// EchoWrapper provides Honeycomb instrumentation for the Echo router via middleware
+type (
+	EchoWrapper struct {
+		handlerNames map[string]string
+		once         sync.Once
+	}
+)
+
+// New returns a new EchoWrapper struct
+func New() *EchoWrapper {
+	return &EchoWrapper{}
+}
+
+// Middleware returns an echo.MiddlewareFunc to be used with Echo.Use()
+func (e *EchoWrapper) Middleware() echo.MiddlewareFunc {
+	return func(next echo.HandlerFunc) echo.HandlerFunc {
+		return func(c echo.Context) error {
+			r := c.Request()
+			// get a new context with our trace from the request
+			ctx, span := common.StartSpanOrTraceFromHTTP(r)
+			defer span.Send()
+			// push the context with our trace and span on to the request
+			c.SetRequest(r.WithContext(ctx))
+
+			// get name of handler
+			handlerName := e.handlerName(c)
+			if handlerName == "" {
+				handlerName = "handler"
+			}
+			span.AddField("handler.name", handlerName)
+			span.AddField("name", handlerName)
+
+			// add route related fields
+			span.AddField("route", c.Path())
```

Comment:
@maplebed - As per your suggestion, I added the following `route` fields.  Should this one that contains the matched route be `route.path` instead?

Reply:
That was one suggestion, but I fear it will lead to more confusion (which path? request path? route path? how are they different again?). I also expect that it's getting into bikeshed territory, so am content to just go with what's here for now.

Reply:
Works for me!